lua/nvim-treesitter/ts_utils.lua

```diff
@@ -5,6 +5,57 @@ local utils = require "nvim-treesitter.utils"
 
 local M = {}
 
+-- determines if a contains node b.
+-- @param a the containing node
+-- @param b the node to be contained
+function M.node_contains(a, b)
+  if a == nil or b == nil then
+    return false
+  end
+
+  local start_row, start_col, end_row, end_col = b:range()
+  return M.is_in_node_range(a, start_row, start_col) and
+    M.is_in_node_range(a, end_row, end_col)
+end
```

Comment:
I don't think this is correct since the end position is exclusive Otherwise this function would also be more elegant for the other one. Maybe have a helper function to obtain the char right before `end_row`,`end_col`? 

 It should also be possible to have only one implementation for those two functions. They are two function with completely different names doing the exact same thing. `M,get_node_range` coerces nodes and ranges to ranges.

Reply:
But, wait in the second function you're using vim ranges. Then, the two functions are actually not the same and the one with start_row/col end_row/col is actually inclusive while the node one is not